### PR TITLE
added colour-demosaicing to the conda-forge

### PR DESCRIPTION
**Pull request by AndrewAnnex**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by AndrewAnnex** [comment]
@conda-forge-admin, please ping conda-forge/staged-recipes

File: recipes/colour-demosaicing/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+test:
+  imports:
+    - colour_demosaicing
+  commands:
+    - pip check
+  requires:
+    - pip
+
+about:
+  home: https://www.colour-science.org/
```
Comment (saraedum) [inline recipes/colour-demosaicing/meta.yaml:39]: Shouldn't the home be https://www.colour-science.org/colour-demosaicing/

Comment (saraedum) [reply recipes/colour-demosaicing/meta.yaml:39]: Can you also add a link to their github https://github.com/colour-science/colour-demosaicing and to their documentation, e.g., https://colour-demosaicing.readthedocs.io/en/latest/reference.html?

**Comment by saraedum** [comment]
Otherwise, this looks good to me :)

File: recipes/colour-demosaicing/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
+{% set name = "colour-demosaicing" %}
+{% set version = "0.2.1" %}
```
Comment (ocefpaf) [inline recipes/colour-demosaicing/meta.yaml:2]: Latest version is 0.2.4, if you need to start with 0.2.1 it is fine but if not, upgrading will save you some PRs later.

File: recipes/colour-demosaicing/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/colour-demosaicing-{{ version }}.tar.gz
+  sha256: 8886a101f40eee8056a2762732f783304ba37026b638de9fb16fa8b8a10ac039
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
```
Comment (ocefpaf) [inline recipes/colour-demosaicing/meta.yaml:14]:
```suggestion
  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
```

File: recipes/colour-demosaicing/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  number: 0
+
+requirements:
+  host:
+    - pip
+    - python >=3.6,<3.11
```
Comment (ocefpaf) [inline recipes/colour-demosaicing/meta.yaml:20]: Metadata says 3.8.

```suggestion
    - python >=3.8,<3.11
```

File: recipes/colour-demosaicing/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    - poetry
+  run:
+    - colour-science >=0.4.0
+    - imageio >=2,<3
+    - numpy >=1.19,<2
+    - python >=3.6,<3.11
```
Comment (ocefpaf) [inline recipes/colour-demosaicing/meta.yaml:26]:
```suggestion
    - python >=3.8,<3.11
```

File: recipes/colour-demosaicing/meta.yaml (hunk 2 above)
Comment (AndrewAnnex) [reply recipes/colour-demosaicing/meta.yaml:2]: sounds good!

File: recipes/colour-demosaicing/meta.yaml (hunk 1 above)
Comment (AndrewAnnex) [reply recipes/colour-demosaicing/meta.yaml:39]: okay I will add those using dev_url and doc_url but if there is a different way to do it just make suggested changes

**Comment by AndrewAnnex** [comment]
hmm maybe that isn't needed outside of the feedstock repo, used to re-rendering things on those prs

File: recipes/colour-demosaicing/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,48 @@
+{% set name = "colour-demosaicing" %}
+{% set version = "0.2.4" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/colour-demosaicing-{{ version }}.tar.gz
```
Comment (ocefpaf) [inline recipes/colour-demosaicing/meta.yaml:9]:
```suggestion
  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/colour_demosaicing-{{ version }}.tar.gz
```

**Pull request merged**